hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/details/ArtifactDetailsLayout.java

```diff
@@ -264,19 +276,77 @@ private void addGeneratedColumnButton(final Table table) {
+        final Button deleteIcon = SPUIComponentProvider.getButton(
+                fileName + "-" + UIComponentIdProvider.UPLOAD_FILE_DELETE_ICON, "",
+                i18n.getMessage(UIMessageIdProvider.CAPTION_DISCARD), ValoTheme.BUTTON_TINY + " " + "blueicon", 
+                true, FontAwesome.TRASH_O, SPUIButtonStyleNoBorder.class);
+        deleteIcon.setData(itemId);
+        deleteIcon.addClickListener(event -> confirmAndDeleteArtifact((Long) itemId, fileName));
+        return deleteIcon;
+    }
+
+    private Button getArtifactDownloadButton(final Table table, final Object itemId) {
+        final String fileName = (String) table.getContainerDataSource().getItem(itemId)
+                .getItemProperty(PROVIDED_FILE_NAME).getValue();
+        final Button downloadIcon = SPUIComponentProvider.getButton(
+                fileName + "-" + UIComponentIdProvider.ARTIFACT_FILE_DOWNLOAD_ICON, "",
+                i18n.getMessage(UIMessageIdProvider.ARTIFACT_DOWNLOAD), ValoTheme.BUTTON_TINY + " " + "blueicon", 
+                true, FontAwesome.DOWNLOAD, SPUIButtonStyleNoBorder.class);
+        downloadIcon.setData(itemId);
+        downloadIcon.addClickListener(event -> downloadArtifact((Long) itemId, fileName));
+        return downloadIcon;
+    }
+
+    private void downloadArtifact(final Long id, final String fileName) {
+        final ConfirmationDialog confirmDialog = new ConfirmationDialog(
+                i18n.getMessage("caption.download.artifact.confirmbox"),
+                i18n.getMessage("message.download.artifact", fileName), i18n.getMessage(UIMessageIdProvider.BUTTON_YES),
+                i18n.getMessage(UIMessageIdProvider.BUTTON_NO), yes -> {
+                    if (yes) {
+                        performDownload(id);
+                    }
+                });
+        UI.getCurrent().addWindow(confirmDialog.getWindow());
+        confirmDialog.getWindow().bringToFront();
+    }
+
+    private void performDownload(final Long id) {
+        Optional<Artifact> artifact = this.artifactManagement.get(id);
+        if (artifact.isPresent()) {
+            Optional<AbstractDbArtifact> file = artifactManagement.loadArtifactBinary(artifact.get().getSha1Hash());
+            if (file.isPresent()) {
+                try (ReadableByteChannel readChannel = Channels.newChannel(file.get().getFileInputStream())) {
+                    writeToOutputFile(artifact.get().getFilename(), readChannel);
+                } catch (IOException e) {
+                    uINotification.displayValidationError(i18n.getMessage("message.artifact.download.failure"));
+                    LOG.error("Artifact reading during download caused an exception", e);
+                }
+            }
+        }
+    }
 
+    private void writeToOutputFile(String fileName, ReadableByteChannel readChannel) {
+        String fileDownloadLoc = System.getProperty("user.home") + "\\Downloads\\" + fileName;
```

Comment:
```suggestion
        final String fileDownloadLoc = System.getProperty("user.home") + File.separator + "Downloads" + File.separator + fileName;
```
The backslash as file separator will not work on unix systems.